packages/core-state/src/wallets/wallet-manager.ts

```diff
@@ -129,19 +129,16 @@ export class WalletManager implements State.IWalletManager {
     }
 
     public loadActiveDelegateList(roundInfo: Shared.IRoundInfo): State.IDelegateWallet[] {
+        const delegates: State.IWallet[] = this.buildDelegateRanking(undefined, roundInfo);
```

Comment:
Why this change? Line 325 should be enough.

Reply:
This change is because we want to check against `maxDelegates` with resigned delegates removed. Thus need to call `buildDelegateRanking()` before the check because `allByUsername()` would not filter anything.

Also this change makes the code simpler because it eliminates the call to `allByUsername()` and removes the variable `delegatesWallets`, so one call and one variable less.

Btw what a wretched naming for `maxDelegates`! We actually want the number of delegates to be bigger than that and emit an error if it is *less*. `minDelegates` would be more appropriate, but out of the scope of this PR.